File: app/src/main/java/com/marlon/portalusuario/une/tarifaElect.java
package com.marlon.portalusuario.une;

import java.util.ArrayList;

public class tarifaElect {
    public ArrayList<RangoConsumo> rangosConsumo;

    public tarifaElect() {
        this.rangosConsumo = new ArrayList<>();
        this.rangosConsumo.add(new RangoConsumo(0.0d, 100.0d, 0.33d));
        this.rangosConsumo.add(new RangoConsumo(100.0d, 150.0d, 1.07d));
        this.rangosConsumo.add(new RangoConsumo(150.0d, 200.0d, 1.43d));
        this.rangosConsumo.add(new RangoConsumo(200.0d, 250.0d, 2.46d));
        this.rangosConsumo.add(new RangoConsumo(250.0d, 300.0d, 3.0d));
        this.rangosConsumo.add(new RangoConsumo(300.0d, 350.0d, 4.0d));
        this.rangosConsumo.add(new RangoConsumo(350.0d, 400.0d, 5.0d));
        this.rangosConsumo.add(new RangoConsumo(400.0d, 450.0d, 6.0d));
        this.rangosConsumo.add(new RangoConsumo(450.0d, 500.0d, 7.0d));
        this.rangosConsumo.add(new RangoConsumo(500.0d, 600.0d, 9.2d));
        this.rangosConsumo.add(new RangoConsumo(600.0d, 700.0d, 9.45d));
        this.rangosConsumo.add(new RangoConsumo(700.0d, 1000.0d, 9.85d));
        this.rangosConsumo.add(new RangoConsumo(1000.0d, 1800.0d, 10.8d));
        this.rangosConsumo.add(new RangoConsumo(1800.0d, 2600.0d, 11.8d));
        this.rangosConsumo.add(new RangoConsumo(2600.0d, 3400.0d, 12.9d));
        this.rangosConsumo.add(new RangoConsumo(3400.0d, 4200.0d, 13.95d));
        this.rangosConsumo.add(new RangoConsumo(4200.0d, 5000.0d, 15.0d));
        this.rangosConsumo.add(new RangoConsumo(5000.0d, 9.99999999E8d, 20.0d));
    }

    public ArrayList<RangoConsumo> getRangosConsumo() {
        return this.rangosConsumo;
    }

    public void setRangosConsumo(ArrayList<RangoConsumo> rangosConsumo2) {
        this.rangosConsumo = rangosConsumo2;
    }

    public tarifaElect(ArrayList<RangoConsumo> rangosConsumo2) {
        this.rangosConsumo = rangosConsumo2;
    }

    public int size() {
        return this.rangosConsumo.size();
    }
}

File: app/src/main/java/com/marlon/portalusuario/une/TarifaElect.kt
package com.marlon.portalusuario.une

object TarifaElect {
    val rangosConsumo: List<RangoConsumo> =
        listOf(
            RangoConsumo(0.0, 100.0, 0.33),
            RangoConsumo(100.0, 150.0, 1.07),
            RangoConsumo(150.0, 200.0, 1.43),
            RangoConsumo(200.0, 250.0, 2.46),
            RangoConsumo(250.0, 300.0, 3.0),
            RangoConsumo(300.0, 350.0, 4.0),
            RangoConsumo(350.0, 400.0, 5.0),
            RangoConsumo(400.0, 450.0, 6.0),
            RangoConsumo(450.0, 500.0, 7.0),
            RangoConsumo(500.0, 600.0, 9.2),
            RangoConsumo(600.0, 700.0, 9.45),
            RangoConsumo(700.0, 1000.0, 9.85),
            RangoConsumo(1000.0, 1800.0, 10.8),
            RangoConsumo(1800.0, 2600.0, 11.8),
            RangoConsumo(2600.0, 3400.0, 12.9),
            RangoConsumo(3400.0, 4200.0, 13.95),
            RangoConsumo(4200.0, 5000.0, 15.0),
            RangoConsumo(5000.0, 9.99999999E8, 20.0),
        )

    fun calculateCost(consumptionKwh: Double): Double {
        var remaining = consumptionKwh
        var total = 0.0
        for (rango in rangosConsumo) {
            val rangeSize = rango.finRango - rango.inicioRango
            if (remaining > rangeSize) {
                total += rangeSize * rango.precioRango
                remaining -= rangeSize
            } else {
                total += remaining * rango.precioRango
                break
            }
        }
        return total
    }
}
